src/viking.js

```diff
@@ -1,11 +1,204 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+  this.health = health;
+  this.strength = strength;
+}
+
+Soldier.prototype.attack = function () {
+  return this.strength;
+}
+
+Soldier.prototype.receiveDamage = function (theDamage) {
+  this.health = this.health - theDamage;
+  if (this.constructor.name === "Viking") {
+    if (this.health > 0) {
+      return `${this.name} has received ${theDamage} points of damage`;
+    } else {
+      // remove from array vikingArmy
+      // I don't know how to access to war1 whithout the name
+
+      for (var i = 0; i < war1.vikingArmy.length; i++) {
+        if (war1.vikingArmy[i].health <= 0) {
+          war1.vikingArmy.splice(i, 1);
+        }
+      }
+
+      this.battleCry();
+      return `${this.name} has died in act of combat`;
+    }
+  } else if (this.constructor.name === "Saxon") {
+    if (this.health > 0) {
+      return `A Saxon has received ${theDamage} points of damage`;
+    } else {
+      // remove from array saxonArmy
+      // I don't know how to access to war1 whithout the name
+
+      for (var i = 0; i < war1.saxonArmy.length; i++) {
+        if (war1.saxonArmy[i].health <= 0) {
+          war1.saxonArmy.splice(i, 1);
+        }
+      }
+
+      this.battleCry();
+      return `A Saxon has died in act of combat`;
+    }
+  }
+
+  if (this.health <= 0) {
+    return this.battleCry();
+  }
+}
+
+Soldier.prototype.attack = function () {
+  return this.strength;
+}
+
+Soldier.prototype.battleCry = function () {
+  return "Odin Owns You All!";
+}
+
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+  this.name = name;
+  Soldier.call(this, health, strength);
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+
+
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+  Soldier.call(this, health, strength);
+}
+
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor = Saxon;
+
+
 
 // War
-function War() {}
+function War() {
+  this.vikingArmy = [];
+  this.saxonArmy = [];
+}
+
+War.prototype.addViking = function (viking) {
+  this.vikingArmy.push(viking);
+}
+
+War.prototype.addSaxon = function (saxon) {
+  this.saxonArmy.push(saxon);
+}
+
+War.prototype.vikingAttack = function () {
+  var randomSaxonNumber = Math.floor(Math.random() * this.saxonArmy.length);
```

Comment:
The code for this method and the method below are very similar. Maybe you could encapsulate them in a function and reuse it